### ISSUE TITLE
Allow users change password via UI

### ISSUE THREAD

**Issue by juanpicado** [verdaccio/verdaccio#401]
#### My reason:

Currently, there is no such interface, any user should be able to update the password. 

####  Config file:	

No config changes need it.

**Comment by Meeeeow** [comment]
Suggest: implement as plugin interface

**Comment by juanpicado** [comment]
Any suggestion about the implementation?

**Comment by juanpicado** [comment]
Because I think we need allow also plugins to support such methods. It's not only UI related, or I'm wrong?

**Comment by Meeeeow** [comment]
You are right, maybe blocked on #169 ? I'm not sure it's a huge change to plugin system or just minor improve

**Comment by juanpicado** [comment]
I haven't check. But I guess minor improve. We can ship that in `2.x` I guess.

**Comment by rodriguesbreno** [comment]
I would suggest go a little further and rethink the Verdaccio UI. 
Then include an user profile section where info can be updated and in the future could support more stuff like data visualizations etc...

Current design is old (from good and old Synopia) and also doesn't match the brand changes yet.
Build on top of it will be just postpone the work that eventually must be done.

**Comment by juanpicado** [comment]
Could you help us to visualize how that idea would look like?

**Comment by jk-jx** [comment]
Suggestion: Create user role for user. Admin and member.

Reason: Because of whoever also can create their own account and publish whatever they want. If you could, please create an admin role which only admin allow to create username and password for user. And if user didn’t own an account he/she could not publish anything to the verdaccio and it will render error msg say that (you didn’t own an account <= sample error msg)

**Comment by rodriguesbreno** [comment]
@juanpicado Sounds good, I'll provide some concepts

**Comment by ahoracek** [comment]
@juanpicado I can help with the roles concept if still needed

**Comment by juanpicado** [comment]
@ahoracek your help is very welcome. We need to trace a plan about what we need to achieve this feature. Could you trace such plan?

**Comment by ahoracek** [comment]
@juanpicado if you already have an idea how user/roles should work (e.g. whether it should be fully on auth plugins or part of core and only auth part implemented as plugins) let me know, based on your input I can prepare plan of what needs to be done. Where should I put the this?

**Comment by juanpicado** [comment]
I think it should be part of the core. 

This is my mental quick roadmap:

- [ ] Let's integrate your functionality into the core https://github.com/verdaccio/verdaccio/issues/392
 creating API enpoints `'/-/npm/v1/user'` https://github.com/ahoracek/verdaccio-profile-api/blob/master/index.js#L102 into the core API endpoints
- [ ]  Allow other auth plugins to implement such funtionality, for this I think you have integrate part of your logic into https://github.com/verdaccio/verdaccio-htpasswd (this is the hardest part )
- [ ] Unit testing and functional testing
- [ ]  Pretty likelike you might need to update Flow types
- [ ]  Create web API endpoints for
- [ ]  Create components and UI implementation in React

It this fine for you?

We should create a ticket (per repository) for each of them an reference them here and one single PR that group all changes + any `@verdaccio/xxx` update.

I can help you allong this process and work on some of the ticket, either UI or unit testing, ask whenever you need help.

**Comment by ahoracek** [comment]
That’s perferct. I will start later today

**Comment by ayusharma** [comment]
@ahoracek If you have any doubt or need any help in https://github.com/verdaccio/verdaccio-htpasswd Please let me know. 

Also, I am creating a feature issue for this https://github.com/verdaccio/verdaccio-htpasswd/issues/32

Thanks,

**Comment by juanpicado** [comment]
@ahoracek any update on this?

**Comment by ahoracek** [comment]
@juanpicado Very little progress recently as I was very busy with my daily work. I get back to this at the end of the week. Sorry for the delay...

**Comment by juanpicado** [comment]
@priscilawebdev will take care of UI and I'll provide all required for UI #392 and https://github.com/verdaccio/verdaccio-htpasswd/pull/38
